hibernate-core/src/main/java/org/hibernate/boot/internal/SessionFactoryOptionsBuilder.java

```diff
@@ -376,6 +381,38 @@ public SessionFactoryOptionsBuilder(StandardServiceRegistry serviceRegistry, Boo
 				CurrentTenantIdentifierResolver.class,
```

Comment:
This should IMO be resolved as bean as well i.e.

```java
		final BeanContainer beanContainer = serviceRegistry.requireService( ManagedBeanRegistry.class ).getBeanContainer();
		if ( beanContainer != null ) {
			this.currentTenantIdentifierResolver = strategySelector.resolveStrategy(
				CurrentTenantIdentifierResolver.class,
				configurationSettings.get( MULTI_TENANT_IDENTIFIER_RESOLVER ),
				null,
				clazz -> beanContainer.getBean( 
					clazz, 
					new BeanContainer.LifecycleOptions() {
						@Override
						public boolean canUseCachedReferences() {
							return true;
						}

						@Override
						public boolean useJpaCompliantCreation() {
							return false;
						}
					}, 
					FallbackBeanInstanceProducer.INSTANCE
				)
			);
		}
		else {
			this.currentTenantIdentifierResolver = strategySelector.resolveStrategy(
				CurrentTenantIdentifierResolver.class,
				configurationSettings.get( MULTI_TENANT_IDENTIFIER_RESOLVER )
			);
		}
```

Reply:
In todays Hibernate ORM team meeting, we decide to hold off on requesting actual beans from the container until JPA 4 when the question of CDI bean manager usability is settled.